mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -4262,9 +4305,43 @@ async fn get_max_volume(ctx: &MmArc, my_coin: &MmCoinEnum, other_coin: &MmCoinEn
     try_s!(check_other_coin_balance_for_swap(ctx, other_coin, None, other_coin_trade_fee).await);
     // calculate max maker volume
     // note the `calc_max_maker_vol` returns [`CheckBalanceError::NotSufficientBalance`] error if the balance of `base_coin` is not sufficient
-    Ok(try_s!(
-        calc_max_maker_vol(ctx, my_coin, &my_balance, FeeApproxStage::OrderIssue).await
-    ))
+    Ok(CoinVolumeInfo {
+        volume: try_s!(calc_max_maker_vol(ctx, my_coin, &my_balance, FeeApproxStage::OrderIssue).await),
+        balance: my_balance,
+    })
+}
+
+pub async fn check_balance_update_loop(ctx: MmWeak, ticker: String, balance: Option<BigDecimal>) {
+    let mut current_balance = balance;
+    loop {
+        Timer::sleep(BALANCE_REQUEST_INTERVAL).await;
+        let ctx = match MmArc::from_weak(&ctx) {
+            Some(ctx) => ctx,
+            None => return,
+        };
+
+        let ordermatch_ctx = OrdermatchContext::from_ctx(&ctx).unwrap();
+        if !ordermatch_ctx
+            .makerorders_ctx
+            .lock()
+            .coin_has_active_maker_orders(&ticker)
+        {
+            ordermatch_ctx.makerorders_ctx.lock().remove_balance_loop(&ticker);
```

Comment:
Please lock `makerorders_ctx` only once here, as it may also cause potential race conditions.